src/insights.test.ts

```diff
@@ -0,0 +1,112 @@
+import { AlgoliaInsights } from './insights';
+import type { InsightsAdditionalEventParams } from './insightsAPIBeaconClient';
+
+describe('insights', () => {
+  describe('uses custom credentials', () => {
+    const additionalParams: InsightsAdditionalEventParams = {
+      headers: {
+        'X-Algolia-Application-Id': 'overrideApp123',
+        'X-Algolia-API-Key': 'overrideKey123',
+      },
+    };
+
+    function assert<
+      TFunction extends keyof Omit<AlgoliaInsights, 'initialized'>
+    >(
+      methodName: TFunction,
+      payload: Parameters<AlgoliaInsights[TFunction]>[0]
+    ) {
+      const insights = new AlgoliaInsights([['init', 'app123', 'key123']]);
+      insights.setUserToken('usertoken123');
+
+      // @ts-expect-error
+      insights[methodName](payload, additionalParams);
+
+      expect(fetch).toHaveBeenLastCalledWith(
+        expect.any(String),
+        expect.objectContaining({
+          headers: expect.objectContaining(additionalParams.headers),
+        })
+      );
+    }
```

Comment:
I'm not fond of abstracting too much in test files because it makes them harder to read and see test cases that closely look like the code a user would write.

Right now, I have to squint to understand what's the actual public API.